api/v1beta2/flowcollector_types.go

```diff
@@ -0,0 +1,769 @@
+/*
```

Comment:
This whole API has received a bunch of changes in the meantime, but since it's a new file it may get unnoticed when you'll rebase, so we need to be careful to not loose anything.
Maybe it will be safer to restart from a fresh copy of v1beta1 and reapply your changes

Reply:
That's right. I'm tried rebasing today but I see a bunch of conflicts that I think it is not worth trying to solve for the sake to time. So I'm about to do exactly that and force update this branch.

Reply:
As you already know it didn't try to solve the conflicts.

So just to let my rebase process registered here (I actually didn't exactly rebased)
- I did a git reset --soft HEAD~3 taking off all and only my commits.
- I stashed those commits out of the working branch
- then rebased with the main branch updated with no errors
- then I opened my stash carefully file by file by using git show stash@{0}:<the file I wanted back> > a diferent name for that file like append a letter b
- then I opened compare selected in VSCode for each and every file
- so I only applied the changes that I actually made letting the new content intact

On the flowcollector_types I had to recopy the v1beta1 and apply my changes after that. If you want to double check, which is highly appreciated, used the compare selected in VSCode with v1beta1 and v1beta2.

Reply:
just did a side by side comparison, looks good!